tests/cli/main.cpp

```diff
@@ -0,0 +1,439 @@
+         BOOST_TEST_MESSAGE("Importing nathan key");
+         BOOST_CHECK_EQUAL(nathan_keys[0], "5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3");
+         BOOST_CHECK(con.wallet_api_ptr->import_key("nathan", nathan_keys[0]));
+      } catch( fc::exception& e ) {
+         edump((e.to_detail_string()));
+         throw;
+      }
+   }
+
+   ~cli_fixture()
+   {
+      BOOST_TEST_MESSAGE("Cleanup cli_wallet::boost_fixture_test_case");
+
+      // wait for everything to finish up
+      fc::usleep(fc::seconds(1));
+
+      app1->shutdown();
+#ifdef _WIN32
+      sockQuit();
+#endif
+   }
+};
+
+///////////////////////////////
+// Tests
+///////////////////////////////
+
+////////////////
+// Start a server and connect using the same calls as the CLI
+////////////////
+BOOST_FIXTURE_TEST_CASE( cli_connect, cli_fixture )
+{
+   BOOST_TEST_MESSAGE("Testing wallet connection.");
+}
+
+BOOST_FIXTURE_TEST_CASE( upgrade_nathan_account, cli_fixture )
+{
+   try
+   {
+      BOOST_TEST_MESSAGE("Upgrade Nathan's account");
+
+      account_object nathan_acct_before_upgrade, nathan_acct_after_upgrade;
+      std::vector<signed_transaction> import_txs;
+      signed_transaction upgrade_tx;
+
+      BOOST_TEST_MESSAGE("Importing nathan's balance");
+      import_txs = con.wallet_api_ptr->import_balance("nathan", nathan_keys, true);
+      nathan_acct_before_upgrade = con.wallet_api_ptr->get_account("nathan");
+
+      BOOST_CHECK(generate_block(app1));
+
+      // upgrade nathan
+      BOOST_TEST_MESSAGE("Upgrading Nathan to LTM");
+      upgrade_tx = con.wallet_api_ptr->upgrade_account("nathan", true);
+
+      nathan_acct_after_upgrade = con.wallet_api_ptr->get_account("nathan");
+
+      // verify that the upgrade was successful
+      BOOST_CHECK_PREDICATE(
+            std::not_equal_to<uint32_t>(),
+            (nathan_acct_before_upgrade.membership_expiration_date.sec_since_epoch())
+                  (nathan_acct_after_upgrade.membership_expiration_date.sec_since_epoch())
+      );
+      BOOST_CHECK(nathan_acct_after_upgrade.is_lifetime_member());
+   } catch( fc::exception& e ) {
+      edump((e.to_detail_string()));
+      throw;
+   }
+}
+
+BOOST_FIXTURE_TEST_CASE( create_new_account, cli_fixture )
+{
+   try
+   {
+      INVOKE(upgrade_nathan_account);
+
+      // create a new account
+      graphene::wallet::brain_key_info bki = con.wallet_api_ptr->suggest_brain_key();
+      BOOST_CHECK(!bki.brain_priv_key.empty());
+      signed_transaction create_acct_tx = con.wallet_api_ptr->create_account_with_brain_key(
+            bki.brain_priv_key, "jmjatlanta", "nathan", "nathan", true
+      );
+      // save the private key for this new account in the wallet file
+      BOOST_CHECK(con.wallet_api_ptr->import_key("jmjatlanta", bki.wif_priv_key));
+      con.wallet_api_ptr->save_wallet_file(con.wallet_filename);
+
+      // attempt to give jmjatlanta some CORE
+      BOOST_TEST_MESSAGE("Transferring bitshares from Nathan to jmjatlanta");
```

Comment:
I would say core tokens instead of bitshares in the message.

Reply:
thanks, changed at https://github.com/peerplays-network/peerplays/pull/98